### ISSUE TITLE
[R-package] duplicated return values in docs

### ISSUE THREAD

**Issue by StrikerRUS** [microsoft/LightGBM#3739]
There are duplicated lines in return statements in our rendered [R-package documentation](https://lightgbm.readthedocs.io/en/latest/R/reference/index.html). For instance,

https://lightgbm.readthedocs.io/en/latest/R/reference/slice.html#value
![image](https://user-images.githubusercontent.com/25141164/103945284-06e28080-5146-11eb-8462-4c993a921416.png)

While normal return statement should look like the following
https://lightgbm.readthedocs.io/en/latest/R/reference/lgb.Dataset.construct.html#value
![image](https://user-images.githubusercontent.com/25141164/103944866-596f6d00-5145-11eb-8c77-6ab1cb8a6dd5.png)

I might be wrong (@jameslamb can correct me), but I believe this happens due to duplicated `@return` directives in the source files.
https://github.com/microsoft/LightGBM/blob/c2dd54f13a58072090e8aaf7dd653f5c5ec1bc22/R-package/R/lgb.Dataset.R#L955
https://github.com/microsoft/LightGBM/blob/c2dd54f13a58072090e8aaf7dd653f5c5ec1bc22/R-package/R/lgb.Dataset.R#L973

To close this issue one should make `pkgdown` (library our R-package documentation is built by) render return value only once for each function/method.

**Comment by jameslamb** [comment]
yes you're right, thanks! Removing the extra `@ return` and then regenerating the docs would fix it. To whoever picks this up, you can regenerate the documentation like this.

```shell
sh build-cran-package.sh
R CMD INSTALL --with-keep.source lightgbm_*.tar.gz
cd R-package
Rscript -e "roxygen2::roxygenize(load = 'installed')"
```

**Comment by zyberg2091** [comment]
can you provide any helpful link to generate docs like this or could you explain more about the process?

**Comment by jameslamb** [comment]
The documentation for the R package is written with [`{roxygen2}`](https://cran.r-project.org/web/packages/roxygen2/vignettes/roxygen2.html), and you can follow the steps in https://github.com/microsoft/LightGBM/issues/3739#issuecomment-756397609 to regenerate them.

**Comment by StrikerRUS** [comment]
Unfortunately, #3850 addressed this issue only partly. Reopening.

**Comment by StrikerRUS** [comment]
- [ ] https://lightgbm.readthedocs.io/en/latest/R/reference/dimnames.lgb.Dataset.html
- [ ] https://lightgbm.readthedocs.io/en/latest/R/reference/getinfo.html
- [ ] https://lightgbm.readthedocs.io/en/latest/R/reference/setinfo.html

**Comment by StrikerRUS** [comment]
Fixed via #3955.
